src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -333,23 +333,38 @@ private void LookupForMatchingGitHubRepository(BuildData mergedBuild)
 
             using (var client = new HttpClient())
             {
-                string[] segments = mergedBuild.AzureDevOpsRepository.Split('/');
-                string repoName = segments[segments.Length - 1];
-                int index = repoName.IndexOf('-');
+                string repoIdentity = string.Empty;
+                string gitHubHost = "github.com";
+
+                if (!Uri.TryCreate(mergedBuild.AzureDevOpsRepository, UriKind.Absolute, out Uri repoAddr))
+                {
+                    throw new Exception($"Can't parse the repository URL: {mergedBuild.AzureDevOpsRepository}");
+                }
+
+                if (repoAddr.Host.Equals(gitHubHost, StringComparison.OrdinalIgnoreCase))
+                {
+                    repoIdentity = repoAddr.AbsolutePath.Trim('/');
+                }
+                else
+                {
+                    string[] segments = mergedBuild.AzureDevOpsRepository.Split('/');
```

Comment:
Is it appropriate to have this named `AzureDevOpsRepository` when it can refer to Azure or GitHub repositories?

Reply:
In a "traditional" scenario were all official builds are building bits from AzDO repos this would always be an `AzureDevOpsRepository` and given that in only a few cases (and once the mirrors and all that discussed on email is worked out there won't be in theory none) this would be a GitHub repo I think the name is ok. But if you have suggestions let's hear them! :)

Reply:
You just said what I was thinking. I also: 1) goal here was just go get builds unblocked asap; 2) we'll need a more thorough refactoring if we assume builds can have Github source repos.

Reply:
Definitely agree on (1) being a priority. Mostly wondering out loud whether names like this could be causing confusion for future changes.